products/identity/Products/Sign-In Service/Engineering Docs/Authentication Types/Service Account Auth (STS)/configuration/staging_prod_setup.md

```diff
@@ -0,0 +1,120 @@
+# Service Account Config Provisioning
+
+## Introduction
+
+This document serves as a guide for Identity engineers implementing a new Service Account Configuration to enable Service Account Auth (STS) for a new client. More details on Service Account Configs themselves can be found [here](./service_account.md).
+
+### Argo Access & `vets-api` Rails console
+
+This guide assumes the user has access to the [vets-api staging & production Argo applications](https://argocd.vfs.va.gov/applications?search=vets-api); all instructions are to be carried out within a Rails console of the `vets-api` instance.
+
+In the "App Details" view of the selected environment there will be one or more `vets-api-web-xxx` "pods", click on any of them to access that pod, then select the "Terminal" tab below the pod name. From the terminal you can enter `bundle exec rails c` to access that environment's Rails instance and perform the necessary commands.
+
+Leaving the terminal window open for an extended period will eventually cause it to lose its connection and force you to close and re-open a pod and terminal session - it is best to prepare the config sequence in a separate editor so that you can quickly input the commands in a fresh pod terminal.
+
+## Creating a Public Certificate
+
+The created Service Account Config will need to include a public certificate in order to validate the service account assertion sent by the STS client.
+
+### Creating an RSA Key
+
+First, locate the private key specified in the configuration instructions. If the new STS config is called by `vets-api`, you will use the SiS `sts_client` key, found at `Settings.sign_in.sts_client.key_path`.
+
+Places Keys are found
+
+* `Settings.sign_in.sts_client.key_path` - self-STS calls
+* other places?
+
+In your Rails console, open the key file and use it to create a new `OpenSSL::PKey::RSA` private key object.
+
+```ruby
+key_file = File.read('Settings.key_path')
+rsa_key = OpenSSL::PKey::RSA.new(key_file)
+```
+
+### Signing a Certificate
+
+Next you will create a certificate and use `rsa_key.public_key` to set its `public_key` attribute. Copy and enter the following into your Rails console, making sure to change the `cert.subject` to reflect `staging.va.gov` if you are creating a staging configuration.
+
+```ruby
+cert = OpenSSL::X509::Certificate.new
```

Comment:
For creating the key/cert, we'll probably want to recommend creating it in Venafi instead of the Rails console. @joeniquette can probably help with a set of instructions on creating the key/cert pair, storing them properly, having the right format for the STS config

Reply:
Sounds good, I'll find a time to go over the `Creating a Public Certificate` section with Joe.

Reply:
[Here](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/teams/vsp/teams/Identity/Documentation/Security/Venafi%20Create%20and%20download%20TLS%20certificates.pdf) are instructions on how to create certs in venafi.